docs/en/lfs.md

```diff
@@ -208,8 +147,12 @@ do
 end
 ```
 
-If you then create a file, say `LFS_dummy_strings.lua`, and put these `local preload` lines in it, and include this file in your `luac.cross -f`, then the cross compiler will generate a ROM string table that includes all strings referenced in this dummy module. You never need to call this module; just it's inclusion in the LFS build is enough to add the strings to the ROM table. Once in the ROM table, then you can use them subsequently in your application without incurring any RAM or LGC overhead.
-A useful starting point may be found in [lua_examples/lfs/dummy_strings.lua](../../lua_examples/lfs/dummy_strings.lua).
+You can do this at the interactive prompt or call it as a debug function during a running application in order to generate this string list, (but note that calling this still creates the overhead of an array in RAM, so you do need to have enough "head room" to do the call).
+
+You can then create a file, say `LFS_dummy_strings.lua`, and insert these `local preload` lines into it.  By including this file in your `luac.cross` compile, then the cross compiler will also include all strings referenced in this dummy module in the generated ROM string table.  Note that you don''t need to call this module; it's inclusion in the LFS build is enough to add the strings to the ROM table. Once in the ROM table, then you can use them subsequently in your application without incurring any RAM or LGC overhead.  A useful starting point may be found in [lua_examples/lfs/dummy_strings.lua](https://github.com/nodemcu/nodemcu-firmware/tree/dev/lua_examples/lfs/dummy_strings.lua); this saves about 4Kb of RAM by moving a lot of common compiler and Lua VM strings into ROM.
+
+Another good use of this technique is when you have resources such as CSS, HTML and JS fragments that you want to output over the internet.  Instead of having lots of small resource files, you can just use string assignments in an LFS module and this will keep these constants in LFS instead.
+ 
```

Comment:
Presumably these should not be `local` in LFS?

Reply:
Not sure that you mean, Nathaniel.  

The preload example just has one or more assignment of a single local to a list of strings.  Do a `luac.cross -l` to see what this generates.  The binary consists of a bunch of `LOADK` instructions, one per string and references a bunch of string constants which get put into the `ROstrt`.  

if you are referring to a resources file then the best way is to avoid creating global and avoid any unnecessary RAM hit by just having a load of:
```Lua
if arg == 'resourceA' then return [[
some long resource
]] end
```

Reply:
I was thinking you meant that the developer should install into LFS a module that looked like this

```lua
resourceA = [[ ... ]]
resourceB = [[ ... ]]
```

rather than like this

```lua
local resourceA = [[ ... ]]
local resourceB = [[ ... ]]
```

since while the latter puts the strings in LFS, it doesn't make it easy to get at them.

Reply:
If your routine is `LFS.getResource`, then this approach would generate a load of globals whether you need them or not.  If you really want to save RAM then in my approach you would do something like:
```Lua
local resource =  LFS.getResource
-- ...
local js = resource('script1')
```
and you could avoid creating any globals.  Or if you have logic pools of resources then each  `getResource()` name could return a small table of resources. 

But the important point is that Lua developers are free to do this in whatever way they feel comfortable and suits their applications.

Reply:
I see.  Thanks for clarification.